### PR TITLE
warp 0.0.3 (new formula)

### PR DESCRIPTION
**Pull request by spolu**
Created manually following:
- [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
- [Formula Cookbook](https://github.com/Homebrew/brew/blob/master/docs/Formula-Cookbook.md)

### CODE REVIEW

**Comment by spolu** [comment]
Looking like a transient failure... will try to push a different sha to trigger a rebuild.

File: Formula/warp.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class Warp < Formula
+  desc "Secure terminal sharing with one simple command"
+  homepage "https://github.com/spolu/warp"
+  url "https://github.com/spolu/warp.git",
+    :tag => "v0.0.3",
+    :revision => "8a04cd7c95973e168a9bf2d2db6522574302164d"
```
Comment (fxcoudert) [inline Formula/warp.rb]: We prefer a `.tar.gz` (e.g. https://github.com/spolu/warp/archive/v0.0.3.tar.gz) to a git tag

File: Formula/warp.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    ENV["GOHOME"] = buildpath
+
+    mkdir_p buildpath/"src/github.com/spolu/"
+    ln_sf buildpath, buildpath/"src/github.com/spolu/warp"
+
+    system "go", "build", "-o", "#{bin}/warp", "github.com/spolu/warp/client/cmd/warp"
+  end
+
+  test do
+    system bin/"warp", "help"
```
Comment (fxcoudert) [inline Formula/warp.rb]: Can we do a test that does more than just output the help?

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/warp.rb (hunk 1 above)
Comment (spolu) [inline Formula/warp.rb]: I'll update that 👍

File: Formula/warp.rb (hunk 2 above)
Comment (spolu) [inline Formula/warp.rb]: So ~anything else would rely on our servers being up and responding. Which is fine, if you think that's better?

**Comment by spolu** [comment]
@fxcoudert PTAL

File: Formula/warp.rb (hunk 2 above)
Comment (bfontaine) [inline Formula/warp.rb]: It’s fine.

File: Formula/warp.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  homepage "https://github.com/spolu/warp"
+  url "https://github.com/spolu/warp/archive/v0.0.3.tar.gz"
+  sha256 "9079899099b63ee470e693b7f2723cbf8d75cafe880b4329cf9f516c2669e8ca"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOBIN"] = bin
+    ENV["GOPATH"] = buildpath
+    ENV["GOHOME"] = buildpath
```
Comment (bfontaine) [inline Formula/warp.rb]: Why do you need to set `GOBIN` and `GOHOME`?

Comment (spolu) [inline Formula/warp.rb]: To be perfectly honest I cargo-culted these ones, let me try to remove them

File: Formula/warp.rb (hunk 2 above)
Comment (spolu) [inline Formula/warp.rb]: @bfontaine fine as in preferable to have a more complex test that relies on our server or fine as in the current state is OK?

Comment (fxcoudert) [inline Formula/warp.rb]: > preferable to have a more complex test that relies on our server

that

Comment (spolu) [inline Formula/warp.rb]: Sounds good! I'll work towards that. Thanks!

**Comment by spolu** [comment]
r? @fxcoudert @bfontaine PTAL

Tests updated with a more meaningful test. Success means that warp is able to put the local terminal in raw mode and prepare everything for a session and attempts to connect to a random inexistent warp ID and exits properly with the expected error coming from the server.

**Comment by fxcoudert** [comment]
Test is failing on CI:

```
==> /usr/local/Cellar/warp/0.0.3/bin/warp connect test-vhhwmvprwlmdyoqi
Error: warp: failed
Expected /\[Error\] Received warp_unknown/ to match "Connected to warp: test-vhhwmvprwlmdyoqi\n[Error] Not running in a terminal.\n".
```

can you update the test to match that message too?

**Comment by spolu** [comment]
@fxcoudert oops. Fixed!

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @spolu for your contribution to Homebrew!

**Comment by spolu** [comment]
\o/

**Pull request closed without merging**